stl/inc/bitset

```diff
@@ -589,13 +589,13 @@ basic_istream<_Elem, _Tr>& operator>>(basic_istream<_Elem, _Tr>& _Istr, bitset<_
             if (_Tr::eq_int_type(_Tr::eof(), _Meta)) { // end of file, quit
                 _State |= _Istr_t::eofbit;
                 break;
-            } else if ((_Char = _Tr::to_char_type(_Meta)) != _Elem0 && _Char != _Elem1) {
+            } else if (!_Tr::eq((_Char = _Tr::to_char_type(_Meta)), _Elem0) && !_Tr::eq(_Char, _Elem1)) {

```

Comment:
This line was a mess before, and adding to it hasn't improved readability. Could we drop the `else` (note that the `if` branch ends with `break`, so the `else` is extraneous) and move the assignment to its own line? (I wouldn't mind removing the other occurrences of `else`-after-`break`, but I won't demand it either.)

Reply:
Agreed, I'll validate and push changes. In new code we almost always avoid mixing modifications and tests. I think this is worth mixing a cleanup into a correctness change (but messing with the other else-breaks, not so much).